src/libertem/api.py

```diff
@@ -1379,6 +1384,12 @@ def _create_local_executor(self):
     def close(self):
         self.executor.close()
 
+    def set_progress_reporter(
+                self,
+                reporter: Union['ProgressReporter', Type['ProgressReporter']]
+            ):
+        self._progress_reporter = reporter
```

Comment:
As this is a method on the `Context`, which is the API most users will be using, can we have a minimal docstring here? I think it's fine to describe that it is an advanced use-case meant for integration, and maybe that the usual way to get a progress bar is to pass `progress=True` to `Context.run_udf`, to redirect users that are lost and believe this is for the common use case.

Stylistic suggestion for consistency: in such a multi-line argument list scenario, I try to keep the closing parenthesis at the same indentation level as the `def`, something like this:

```suggestion
    def set_progress_reporter(
        self,
        reporter: Union['ProgressReporter', Type['ProgressReporter']]
    ):
        self._progress_reporter = reporter
```

(at some point, we should switch to `black` for code formatting I guess)

Reply:
I was in the process of writing the docstring when I decided to write the comment for `(..., progress: bool | ProgressReporter)`. Do you want to retain both approaches then?

Reply:
> I was in the process of writing the docstring when I decided to write the comment for `(..., progress: bool | ProgressReporter)`. Do you want to retain both approaches then?

Yeah, I wrote this comment before I saw yours. I'm fine with having only one approach, i.e. `run_udf`.